alma/lib/ShareOfCheckout/ShareOfCheckoutHelper.php

```diff
@@ -236,21 +236,29 @@ public function removeConsent()
      */
     public function handleCheckoutConsent($consentAttribute)
     {
-        $isUserConsent = (bool) Tools::getValue($consentAttribute);
-        $isSOCActivated = Configuration::get(ShareOfCheckoutAdminFormBuilder::ALMA_ACTIVATE_SHARE_OF_CHECKOUT);
+        $userConsent = Tools::getValue($consentAttribute);
+        $timeConsent = Settings::getCurrentTimestamp();
+        
+        if (Settings::getTimeConsentShareOfCheckout()) {
+            $timeConsent = Settings::getTimeConsentShareOfCheckout();
+        }
 
         try {
-            if ($isUserConsent && !$isSOCActivated) {
-                // we need to activate share of checkout
-                $this->addConsent();
-                Settings::updateValue(ShareOfCheckoutAdminFormBuilder::ALMA_ACTIVATE_SHARE_OF_CHECKOUT, true);
-                Settings::updateValue(ShareOfCheckoutAdminFormBuilder::ALMA_SHARE_OF_CHECKOUT_DATE, Settings::getCurrentTimestamp());
-            }
-            if (!$isUserConsent && $isSOCActivated) {
-                // we need to desactivate share of checkout
-                $this->removeConsent();
-                Settings::updateValue(ShareOfCheckoutAdminFormBuilder::ALMA_ACTIVATE_SHARE_OF_CHECKOUT, '');
-                Settings::updateValue(ShareOfCheckoutAdminFormBuilder::ALMA_SHARE_OF_CHECKOUT_DATE, null);
+            if (Settings::isShareOfCheckoutSetting()) {
+                if ($userConsent == ShareOfCheckoutAdminFormBuilder::ALMA_SHARE_OF_CHECKOUT_CONSENT_YES) {
```

Comment:
to many if/else complexity, maybe move the code in the first if( if (Settings::isShareOfCheckoutSetting()) ) in an other function